### PR TITLE
Savings module: deposit querier

### PR DESCRIPTION
**Pull request by denalimarsh**
Querier for the saving's module deposits

Base branch should be changed to master before merge.

### CODE REVIEW

**Comment by denalimarsh** [comment]
Removed `types/querier.go`, `keeper/querier.go` and `client/rest/query.go`/`client/rest/rest.go` due to [discussion](https://github.com/Kava-Labs/kava/pull/1188#issuecomment-1072697814) in the scaffolding PR. Should the remaining files in `client/rest` be deleted as well?

File: x/savings/keeper/grpcquery_test.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	tmprototypes "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/kava-labs/kava/app"
+	"github.com/kava-labs/kava/x/savings/keeper"
+	"github.com/kava-labs/kava/x/savings/types"
+)
+
+type grpcQueryTestSuite struct {
```
Comment (drklee3) [inline x/savings/keeper/grpcquery_test.go:16]: test suite doesn't seem to be actually run, missing 

```go
func TestGrpcQueryTestSuite(t *testing.T) {
	suite.Run(t, new(grpcQueryTestSuite))
}
```

File: x/savings/keeper/grpc_query.go [hunk 2]
```diff
@@ -33,3 +36,69 @@ func (s queryServer) Params(c context.Context, req *types.QueryParamsRequest) (*
 
 	return &types.QueryParamsResponse{Params: params}, nil
 }
+
+func (s queryServer) Deposits(ctx context.Context, req *types.QueryDepositsRequest) (*types.QueryDepositsResponse, error) {
+	if req == nil {
+	}
```
Comment (drklee3) [inline x/savings/keeper/grpc_query.go:42]: missing return err here

File: x/savings/keeper/grpcquery_test.go (hunk 1 above)
Comment (denalimarsh) [reply x/savings/keeper/grpcquery_test.go:16]: Good catch

**Review by karzak — APPROVED** [review]
+1 to Derrick's suggestions. Functionality looks good/correct with kvtool - LGTM after making Derrick's suggested changes.

File: x/savings/keeper/grpcquery_test.go (hunk 1 above)
Comment (denalimarsh) [reply x/savings/keeper/grpcquery_test.go:17]: Updated in Updated in https://github.com/Kava-Labs/kava/pull/1193/commits/a944cc35ceeb67a5c57c41b3d59bc1231bd5c6c8

File: x/savings/module.go [hunk 3]
```diff
@@ -126,7 +125,9 @@ func (AppModule) QuerierRoute() string {
 
 // LegacyQuerierHandler returns no sdk.Querier.
 func (am AppModule) LegacyQuerierHandler(legacyQuerierCdc *codec.LegacyAmino) sdk.Querier {
-	return keeper.NewQuerier(am.keeper, legacyQuerierCdc)
+	return func(ctx sdk.Context, path []string, req abci.RequestQuery) (res []byte, err error) {
+		return []byte{}, nil
+	}
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/savings/module.go]: Updated in https://github.com/Kava-Labs/kava/pull/1193/commits/a944cc35ceeb67a5c57c41b3d59bc1231bd5c6c8

File: x/savings/keeper/grpc_query.go [hunk 4]
```diff
@@ -33,3 +36,69 @@ func (s queryServer) Params(c context.Context, req *types.QueryParamsRequest) (*
...
+			if deposit.Amount.AmountOf(req.Denom).IsPositive() {
+				deposits = append(deposits, deposit)
+			}
+			return false
+		})
+	default:
+		s.keeper.IterateDeposits(sdkCtx, func(deposit types.Deposit) (stop bool) {
+			deposits = append(deposits, deposit)
+			return false
+		})
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/savings/keeper/grpc_query.go:86]: Decided not to implement this due to the hasOwner/hasDenom - this is the standard in Hard module at the moment

**Pull request merged**